src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<ExpressionSyntax>
+        {
+            private readonly ArrayBuilder<ExpressionSyntax> _assignmentTargets;
+            private readonly bool _isAllThrowStatements;
+
+            private Rewriter(bool isAllThrowStatements)
+            {
+                _assignmentTargets = ArrayBuilder<ExpressionSyntax>.GetInstance();
+                _isAllThrowStatements = isAllThrowStatements;
+            }
+
+            private void Free()
+            {
+                _assignmentTargets.Free();
+            }
+
+            public static StatementSyntax Rewrite(
+                SwitchStatementSyntax switchStatement, SemanticModel semanticModel, SyntaxEditor editor,
+                SyntaxKind nodeToGenerate, bool shouldMoveNextStatementToSwitchExpression)
+            {
+                var rewriter = new Rewriter(isAllThrowStatements: nodeToGenerate == SyntaxKind.ThrowStatement);
+
+                // Rewrite the switch statement as a switch expression.
+                var switchExpression = rewriter.RewriteSwitchStatement(switchStatement,
+                    allowMoveNextStatementToSwitchExpression: shouldMoveNextStatementToSwitchExpression);
+
+                // Only on simple assignments we attempt to remove variable declarators.
+                var isSimpleAssignment = nodeToGenerate == SyntaxKind.SimpleAssignmentExpression;
+                var generateDeclaration = isSimpleAssignment && rewriter.TryRemoveVariableDeclarators(switchStatement, semanticModel, editor);
```

Comment:
>TryRemoveVariableDeclarators [](start = 73, length = 28)

Should we use [IMoveDeclarationNearReferenceService](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/MoveDeclarationNearReference/IMoveDeclarationNearReferenceService.cs,9) here instead of having logic to determine when to remove declarator and generate a new declaration?